python/lsst/analysis/tools/tasks/astrometricCatalogMatch.py

```diff
@@ -0,0 +1,215 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = (
+    "AstrometricCatalogMatchConfig",
+    "AstrometricCatalogMatchTask",
+    "AstrometricCatalogMatchVisitConfig",
+    "AstrometricCatalogMatchVisitTask",
+)
+
+import lsst.geom
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from astropy.table import Table
+from astropy.time import Time
+from lsst.pipe.tasks.configurableActions import ConfigurableActionStructField
+from lsst.pipe.tasks.loadReferenceCatalog import LoadReferenceCatalogTask
+
+from ..actions.vector import VisitPlotFlagSelector
+from ..tasks.catalogMatch import CatalogMatchConfig, CatalogMatchConnections, CatalogMatchTask
+
+
+class AstrometricCatalogMatchConfig(CatalogMatchConfig, pipelineConnections=CatalogMatchConnections):
+    bands = pexConfig.ListField[str](
+        doc="The bands to persist downstream",
+        default=["g", "r", "i", "z", "y"],
+    )
+
+    def setDefaults(self):
+        super().setDefaults()
+        self.referenceCatalogLoader.doApplyColorTerms = False
+        self.referenceCatalogLoader.refObjLoader.requireProperMotion = False
+        self.referenceCatalogLoader.refObjLoader.anyFilterMapsToThis = "phot_g_mean"
+
+
+class AstrometricCatalogMatchTask(CatalogMatchTask):
+    """Match a tract-level catalog to a reference catalog"""
+
+    ConfigClass = AstrometricCatalogMatchConfig
+    _DefaultName = "analysisToolsAstrometricCatalogMatch"
+
+    def runQuantum(self, butlerQC, inputRefs, outputRefs):
+        # Docs inherited from base class
+
+        inputs = butlerQC.get(inputRefs)
+        columns = self.prepColumns(self.config.bands)
+        table = inputs["catalog"].get(parameters={"columns": columns})
+        inputs["catalog"] = table
+
+        tract = butlerQC.quantum.dataId["tract"]
+
+        loaderTask = LoadReferenceCatalogTask(
+            config=self.config.referenceCatalogLoader,
+            dataIds=[ref.dataId for ref in inputRefs.refCat],
+            name=inputs["refCat"][0].ref.datasetType.name,
+            refCats=inputs["refCat"],
+        )
+
+        skymap = inputs.pop("skymap")
+        loadedRefCat = self._loadRefCat(loaderTask, skymap[tract])
+        outputs = self.run(catalog=inputs["catalog"], loadedRefCat=loadedRefCat, bands=self.config.bands)
+
+        butlerQC.put(outputs, outputRefs)
+
+
+class AstrometricCatalogMatchVisitConnections(
+    CatalogMatchConnections,
```

Comment:
I don't think you're getting anything from inheriting from `CatalogMatchConnections`, but you're adding `skymap` which might mess up `run()`? Not sure if it would be a problem.